chain/bitcoind_conn.go

```diff
@@ -370,11 +447,149 @@ func (c *BitcoindConn) blockEventHandler() {
 	}
 }
 
+// blockEventHandlerRPC is a goroutine that uses the rpc client to check if we
+// have a new block every so often.
+func (c *BitcoindConn) blockEventHandlerRPC() {
+	defer c.wg.Done()
+
+	log.Info("Started polling for new bitcoind blocks via RPC.")
+	ticker := time.NewTicker(c.cfg.PollBlockTimer)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			// Every so many seconds, we poll to see if there's a new
+			// block.
+			newBlockHash, err := c.client.GetBestBlockHash()
+			if err != nil || newBlockHash == nil {
+				log.Errorf("Unable to retrieve best block "+
+					"hash: %v", err)
+				continue
+			}
+			newBlockHeader, err := c.client.GetBlockHeaderVerbose(
+				newBlockHash,
+			)
+			if err != nil {
+				log.Errorf("Unable to retrieve block height: "+
+					"%v", err)
+				continue
+			}
+
+			c.rescanClientsMtx.Lock()
+			if c.rescanClients == nil || len(c.rescanClients) < 1 {
+				log.Errorf("No client added yet")
+				c.rescanClientsMtx.Unlock()
+				continue
+			}
+
+			// Grab one of the clients from the rescanClients map
+			// so we can look at its latest block height.
+			var client *BitcoindClient
+			for _, currClient := range c.rescanClients {
+				client = currClient
+				break
+			}
+			c.rescanClientsMtx.Unlock()
+
+			// Check the block height of one of the clients to see
+			// if we're up to date or not.
+			bestBlock, err := client.BlockStamp()
+			if err != nil {
+				log.Errorf("Unable to get most recent block: "+
+					"%v", err)
```

Comment:
I dont think the code needs to be client-aware at all. See how I have done it [here](https://github.com/ellemouton/btcwallet/pull/2).